crates/flux-middle/src/rty/pretty.rs

```diff
@@ -235,6 +236,32 @@ impl Pretty for SubsetTy {
     }
 }
 
+// This is a trick to avoid pretty printing `S [S { x: 10, y: 20}]`
+// and instead just print `S[{x: 10, y: 20}]` for struct-valued indices.
+struct TopIndexExpr(Expr);
```

Comment:
I'd call this `IdxFmt` or `IdxPretty`

```suggestion
struct IdxFmt(Expr);
```